### PR TITLE
add `PrometheusRuleFailures`

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/20510

This PR adds the `PrometheusRuleFailures` alert which will notify us when prometheus rules evaluation fails. This is bad because if prometheus can't evaluate a rule we will never get alerted for it.

i.e. currently failing SLO rules is failing and went unoticed

```
msg="Evaluating rule failed" rule="alert: ServiceLevelBurnRateTooHigh ...
```

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 1]
```diff
@@ -38,3 +38,10 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    - alert: PrometheusRuleFailures
+      annotations:
+        description: {{`Prometheus {{$labels.namespace}}/{{$labels.pod}} has failed to evaluate {{ printf "%.0f" $value }} rules.`}}
+        summary: Prometheus is failing rule evaluations.
+      expr: increase(prometheus_rule_evaluation_failures_total[30m]) > 0
+      labels:
+        severity: critical
```
Comment (JosephSalisbury) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:47]: what does `critical` mean?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:47]: eh sorry, that was a copy paste

Comment (JosephSalisbury) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:47]: I could believe only having this in business hours would be nice too

**Comment by TheoBrigitte** [comment]
Tested and looks alright :heavy_check_mark: 

image

Could not extract the rule group name from labels. And adding something like the following to the description feels a bit much.

```
/etc/prometheus/rules/prometheus-ghost-rulefiles-0/monitoring-service-level.rules-04ac4d70-77e5-4d2c-8e1a-5ee85bd9f529.yaml;service-level
```

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml]: Adjusted labels

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 2]
```diff
@@ -38,3 +38,15 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    - alert: PrometheusRuleFailures
+      annotations:
+        description: {{`Prometheus {{$labels.installation}}/{{$labels.cluster_id}} has failed to evaluate rule(s) {{ printf "%.0f" $value }} time(s).`}}
+        summary: Prometheus is failing rule evaluations.
+      expr: increase(prometheus_rule_evaluation_failures_total[30m]) > 0
```
Comment (JosephSalisbury) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:45]: why `increase` over `rate`

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml]: Yeah good point `rate` should be used an not `increase`.

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 3]
```diff
@@ -38,3 +38,15 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    - alert: PrometheusRuleFailures
+      annotations:
+        description: {{`Prometheus {{$labels.installation}}/{{$labels.cluster_id}} has failed to evaluate rule(s) {{ printf "%.0f" $value }} time(s).`}}
```
Comment (JosephSalisbury) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:43]:
```suggestion
        description: {{`Prometheus has failed to evaluate rule(s) {{ printf "%.0f" $value }} time(s).`}}
```

Comment (JosephSalisbury) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:43]: we already have this info in the title

**Review by JosephSalisbury — APPROVED** [review]
small nit, but lgtm otherwise

**Pull request merged**